packages/typespec-client-generator-core/src/public-utils.ts

```diff
@@ -698,3 +707,61 @@ export function isAzureCoreModel(t: SdkType): boolean {
 export function isPagedResultModel(context: TCGCContext, t: SdkType): boolean {
   return context.__pagedResultSet.has(t);
 }
+
+/**
+ * Find corresponding http parameter list for a client initialization or service method parameter or the property of that parameter.
+ *
+ * @param method
+ * @param param
+ * @returns
+ */
+export function getHttpOperationParameter(
```

Comment:
wonder if the performance would be a concern here?

Reply:
i think it's ok. since we already have the corresponding map, it may take some time but not too much.